src/index.js

```diff
@@ -32,3 +52,12 @@ export function parseExpression(input, options) {
 
 
 export { tokTypes };
+
+function hasModuleSyntax(ast) {
+  return ast.program.body.some((child) =>
+    child.type === "ImportDeclaration" ||
+    child.type === "ExportDeclaration" ||
+    child.type === "ExportAllDeclaration" ||
+    child.type === "ExportDefaultDeclaration"
```

Comment:
You still get an AwaitExpression either way, I think that part of the spec is more to make like easier for parsers if they want.

In this example, if you use `await` as an identifier it'll throw and reparse as a script successfully, which I think is good?